modules/objdetect/src/qrcode.cpp

```diff
@@ -958,35 +1768,828 @@ bool QRCodeDetector::detect(InputArray in, OutputArray points) const
+ * return :output string
+ * */
+std::string show_poly(const Mat& p,OUTPUT o=HEX){
+    std::string s;
+    for(int i=0;i<p.cols;i++){
+        char tmp[10];
+        switch (o){
+            case HEX:
+                sprintf(tmp,"%02X",(int)p.ptr(0)[i]);//
+                break;
+            case OCT:
+                sprintf(tmp,"%d",(int)p.ptr(0)[i]);
+                break;
+            case ALPHA:
+                sprintf(tmp,"%d",gf_log[(int)p.ptr(0)[i]]%255);//%02X
+                break;
+        }
+        s=" "+s;
+        s=tmp+s;
+    }
+    s=s+'\n';
+    return  s;
+}
+/* gf_poly_eval :
+ *      evaluate a polynomial at a particular value of x, producing a scalar result
+ * params @ poly 15bit format_Info,  uint8_t x(a scalar)
+ * return @ result
+
+ * using the Horner's method here:
+ * 01 x4 + 0f x3 + 36 x2 + 78 x + 40 = (((01 x + 0f) x + 36) x + 78) x + 40
+ * doing this by simple addition and multiplication
+ * */
+uint8_t gf_poly_eval(const Mat& poly,uint8_t x){
+    /*Note the calculation begins at the high times of items,
+         * That's to say , start from the large index in Mat
+         * */
+    int index=poly.cols-1;
+    uint8_t y=poly.ptr(0)[index];
+    for(int i =index-1;i>=0;i--){
+        y = gf_mul(x,y) ^ poly.ptr(0)[i];
+    }
+    return y;
+}
+/*
+ * func @  multiply a polynomial by a scalar
+ * */
+Mat gf_poly_scale(const Mat & p,int scalar) {
+    int len = p.cols;
+    Mat r(1,len,CV_8UC1,Scalar(0));
+
+    for(int i = 0; i < len;i++){
+        r.ptr(0)[i] = gf_mul(p.ptr(0)[i], (uint8_t)scalar);
+    }
+    return r;
+}
+/*
+ * func @  "adds" two polynomials (using exclusive-or, as usual).
+ * */
+Mat gf_poly_add(const Mat & p,const Mat & q){
+    int p_len=p.cols;
+    int q_len=q.cols;
+    Mat r (1,max(p_len,q_len),CV_8UC1,Scalar(0));
+
+    //int r_len=r.cols;
+
+    for (int i = 0; i< p_len ;i++){
+        r.ptr(0)[i] = p.ptr(0)[i];
+    }
+    for (int i = 0; i< q_len ;i++){
+        r.ptr(0)[i] ^= q.ptr(0)[i];//+r_len-q_len
+    }
+    return r;
+}
+
+/* multiplication between two polys
+ * params @ poly p , poly q
+ * return @ result poly = p * q
+ */
+/*
+       10001001
+    *  00101010
+ ---------------
+      10001001
+^   10001001
+^ 10001001
+ ---------------
+  1010001111010*/
+Mat gf_poly_mul(const Mat &p,const Mat &q){
+
+    /* multiplication == addition among items*/
+    Mat r(1,p.cols+q.cols-1,CV_8UC1,Scalar(0));
+    int len_p=p.cols;
+    int len_q=q.cols;
+    for(int j = 0; j<len_q;j++) {
+        if(!q.ptr(0)[j])
+            continue;
+        //cout<<"round : "<<j<<endl;
+
+        for (int i = 0; i < len_p; i++) {
+            if(!p.ptr(0)[i])
+                continue;
+            r.ptr(0)[i+j] ^= gf_mul(p.ptr(0)[i], q.ptr(0)[j]);
+        }
+    }
+    return r;
+}
+
+/*gf_poly_div:
+ *   This function is for getting the ECC for the data string ,which is implemented by doing a poly division.
+ * params @ const Mat& dividend,const Mat& divisor
```

Comment:
where do you use doxygen?

Reply:
what does this mean?

Reply:
@ZhengQiushi It looks like you use doxygen syntax for html documentation generation. I think we don’t need documentation for description of all methods of the class, we need it only for public methods that are available to class users. You can see detailed description of objdetect methods with doxygen usage in [objdetect.hpp](https://github.com/opencv/opencv/blob/3.4/modules/objdetect/include/opencv2/objdetect.hpp)